From b90ae75df6f76d2de8a9642c03c5d71634c03030 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Mon, 4 Jan 2016 17:20:02 +0800
Subject: [PATCH 0812/5242] MLK-12103 media: capture: ov5640_mipi: add more
 delay for camera to stay stable

commit  fcc51f4eb0e38b631b1380b114c434dde7097f65 from
https://source.codeaurora.org/external/imx/linux-imx.git

For power saving, the camera is set ON/OFF by 'PWDN' pin.
However, the DQBUF timeout problems have been observed occasionally when
do the resolution switch for the camera. The reason could be that the power
for the camera is set to off then on when do the switch, but the delay time
is not enough for camera to go stable. The patch sets it to an empirical
value and it has passed over 3-day stress test.

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/mxc/capture/ov5640_mipi.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/mxc/capture/ov5640_mipi.c b/drivers/media/platform/mxc/capture/ov5640_mipi.c
index ec813a6..0d146b5 100644
--- a/drivers/media/platform/mxc/capture/ov5640_mipi.c
+++ b/drivers/media/platform/mxc/capture/ov5640_mipi.c
@@ -686,7 +686,7 @@ static void ov5640_standby(s32 enable)
 	else
 		gpio_set_value(pwn_gpio, 0);
 
-	msleep(2);
+	msleep(100);
 }
 
 static void ov5640_reset(void)
-- 
1.7.9.5

